docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,895 @@
+requirements than a whole MeshService
+
+##### Overriding on universal
+
+Since all policies on universal are system policies (we don't have consumer/producer policies as we don't have namespaces),
+our example will look like: 
+
+```yaml
+# route to backend MeshService
+type: MeshHTTPRoute
+name: route-to-backend
+mesh: default
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      rules:
+        - matches:
+            - path:
+                type: PathPrefix
+                value: "/slow-endpoint"
+          default:
+            backendRefs:
+              - kind: MeshService
+                name: backend
+---
+# MeshTimeout targeting whole MeshService
+type: MeshTimeout
+name: timeout-on-backend-service
+mesh: default
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      default: conf1
+---
+# MeshTimeout targeting route
+type: MeshTimeout
+name: timeout-on-backend-route
+mesh: default
+spec:
+  to:
+    - targetRef:
+        kind: MeshHTTPRoute
+        name: route-to-backend
+      default: conf2
+```
+
+To override this configuration for a service consuming this route we have to create policy with `spec.targetRef` MeshSubset: 
+
+```yaml
+# MeshTimeout targeting whole MeshService
+type: MeshTimeout
+name: timeout-on-backend-service-override
+mesh: default
+spec:
+  targetRef:
+    kind: MeshSubset
+    tags:
+      services-group: frontend # custom tag on fronted services
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      default: conf3
+---
+# MeshTimeout targeting route
+type: MeshTimeout
+name: timeout-on-backend-route-override
+mesh: default
+spec:
+  targetRef:
+    kind: MeshSubset
+    tags:
+      services-group: frontend # custom tag on fronted services
+  to:
+    - targetRef:
+        kind: MeshHTTPRoute
+        name: route-to-backend
+      default: conf4
+```
+
+This will result in the same configuration as in previous example, because `MeshSubset` kind is more specific than `Mesh`:
+
+```yaml
+resourceRules:
+  backend:
+    conf: merge($conf1, $conf3)
+  backend-route:
+    conf: merge($conf1, $conf2, $conf3, $conf4)
+```
+
+### Backward compatibility and migration
+
+First we need to start by deprecating putting `Mesh*Route` in `spec.targetRef`.
+At the moment we are building `Rules` object with Subset for matching. This will change with implementation of 
+`policy matching with real resources` MADR. For real resources we will build `ResourceRule` object. To smoothen the transition
+we can still build old `Rules` for old policies and new `ResourceRule` for newly created policies with `Mesh*Route` in `spec.to[].targetRef`.
+At the policy level we can prioritize new `ResourceRule` over old `Rules`. Thanks to this new way of targeting `Mesh*Routes`
+will take precedence, so that users can easily switch to new targeting mechanism without any downtime.
```

Comment:
What about the deprecation around number of entries in `to` for `Mesh*Route` ?

Reply:
Added info about deprecating, and using consumer/producer policcies to smoothen it out